graphene_django/optimization.py

```diff
@@ -51,6 +51,15 @@ def get_related_fetches_for_model(model, graphql_ast):
 
         try:
             related_model = get_related_model(selection_field)
+            if not related_model:
```

Comment:
Let's move this below the `except` and simply set `related_model = None` in the `except` block. Then we won't have duplication of this section.